pkg/dyninst/decode/decoder.go

```diff
@@ -167,7 +169,12 @@ func (d *Decoder) Decode(
 	}
 	b := bytes.NewBuffer(buf)
 	enc := jsontext.NewEncoder(b)
-	numExpressions := len(d.snapshotMessage.Debugger.Snapshot.Captures.Entry.rootType.Expressions)
+	var numExpressions int
+	if d.snapshotMessage.Debugger.Snapshot.Captures.Entry != nil {
+		numExpressions = len(d.snapshotMessage.Debugger.Snapshot.Captures.Entry.rootType.Expressions)
+	} else if d.snapshotMessage.Debugger.Snapshot.Captures.Lines != nil {
+		numExpressions = len(d.snapshotMessage.Debugger.Snapshot.Captures.Lines.capture.rootType.Expressions)
+	}
```

Comment:
It's arguably a bug but we should also add the number of return expressions if return is non nil in the entry case. Writing this as a reminder to me. You don't need to do it here.